### DISCUSSION TITLE
tikv在scan最新插入的数据的时候性能低

### DISCUSSION THREAD

**Discussion by longlong524** [juicedata/juicefs#4827] [Q&A]
【 TiDB 使用环境】生产环境
【 TiDB 版本】
Starting component cluster: /root/.tiup/components/cluster/v1.12.5/tiup-cluster display tidbCluster
Cluster type: tidb
Cluster name: tidbCluster
Cluster version: v7.1.1

【复现路径】tikv作为juicefs的元数据库使用，然后我集成并mount之后，新建了三个文件夹，test20w，test20w2, test20w3，三个文件夹，然后分别在三个文件夹新建20w个文件，之后执行tikv的txn的scan操作，对三个文件夹下的文件进行扫描。发现最后执行新建20w文件命令的文件夹用时最久。如果之后有另外一个文件夹执行了新建大量文件的命令，那么之前的文件夹的scan命令用时就会正常

大神们帮我看看

**Comment by davies** [comment]
这个 scan 是扫描哪种 key？估计跟 tikv 处理新老数据的机制相关，可能到 tikv 的论坛问一下更合适。

**Comment by longlong524** [comment]
就是txn的iter方法，指定一个起始key和endkey

---- 回复的原邮件 ----
| 发件人 | Davies ***@***.***> |
| 发送日期 | 2024年05月13日 10:02 |
| 收件人 | juicedata/juicefs ***@***.***> |
| 抄送人 | Xianglong ***@***.***>,
Author ***@***.***> |
| 主题 | Re: [juicedata/juicefs] tikv在scan最新插入的数据的时候性能低 (Discussion #4827) |

这个 scan 是扫描哪种 key？估计跟 tikv 处理新老数据的机制相关，可能到 tikv 的论坛问一下更合适。
